SDL/Module.cuh

```diff
@@ -105,51 +219,632 @@ namespace SDL
+            host_isLower[index] = isLower;
+
+            //assigning other variables!
+            if(detId == 1)
+            {
+                host_moduleType[index] = PixelModule;
+                host_moduleLayerType[index] = SDL::InnerPixelLayer;
+                host_slopes[index] = 0;
+                host_drdzs[index] = 0;
+                host_isAnchor[index] = false;
+            }
+            else
+            {
+                host_moduleType[index] = ( m_t == 25 ? SDL::TwoS : SDL::PS );
+                host_moduleLayerType[index] = ( m_t == 23 ? SDL::Pixel : SDL::Strip );
+
+                if(host_moduleType[index] == SDL::PS and host_moduleLayerType[index] == SDL::Pixel)
+                {
+                    host_isAnchor[index] = true;
+                }
+                else if(host_moduleType[index] == SDL::TwoS and host_isLower[index])
+                {
+                    host_isAnchor[index] = true;   
+                }
+                else
+                {
+                    host_isAnchor[index] = false;
+                }
+
+                host_slopes[index] = (subdet == Endcap) ? endcapGeometry.getSlopeLower(detId) : tiltedGeometry.getSlope(detId);
+                host_drdzs[index] = (subdet == Barrel) ? tiltedGeometry.getDrDz(detId) : 0;
+            }
+        }
+
+        //partner module stuff, and slopes and drdz move around
+        for(auto it = (*detIdToIndex).begin(); it != (*detIdToIndex).end(); it++)
+        {
+            auto& detId = it->first;
+            auto& index = it->second;
+            if(detId != 1)
+            {
+                host_partnerModuleIndices[index] = (*detIdToIndex)[modulesInGPU->parsePartnerModuleId(detId, host_isLower[index], host_isInverted[index])];
+                //add drdz and slope importing stuff here!
+                if(host_drdzs[index] == 0)
+                {
+                    host_drdzs[index] = host_drdzs[host_partnerModuleIndices[index]];
+                }
+                if(host_slopes[index] == 0)
+                {
+                    host_slopes[index] = host_slopes[host_partnerModuleIndices[index]];
+                }
+            }
+        }
+
+        cudaMemcpyAsync(modulesInGPU->nModules,&nModules,sizeof(uint16_t),cudaMemcpyHostToDevice,stream);
+        cudaMemcpyAsync(modulesInGPU->nLowerModules,&nLowerModules,sizeof(uint16_t),cudaMemcpyHostToDevice,stream);
+        cudaStreamSynchronize(stream);
+
+        cudaMemcpyAsync(modulesInGPU->moduleType,host_moduleType,sizeof(ModuleType)*nModules,cudaMemcpyHostToDevice,stream);
+        cudaMemcpyAsync(modulesInGPU->moduleLayerType,host_moduleLayerType,sizeof(ModuleLayerType)*nModules,cudaMemcpyHostToDevice,stream);
+        cudaStreamSynchronize(stream);
+
+        //alpaka::memcpy(queue, modulesBuf->moduleType_buf, moduleType_buf, nModules);
+        //alpaka::memcpy(queue, modulesBuf->moduleLayerType_buf, moduleLayerType_buf, nModules);
```

Comment:
Alpaka does not produce correct behavior for memcpy on these two buffers with custom datatypes. Leaving this for the next PR.

Reply:
I was able to fix this by moving to a higher version of Alpaka and not specifying the number of elements.